airbyte-integrations/connectors/source-zendesk-sell/source_zendesk_sell/source.py

```diff
@@ -0,0 +1,295 @@
+#
+# Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+#
+
+
+from abc import ABC
+from typing import Any, Iterable, List, Mapping, MutableMapping, Optional, Tuple
+import re
+import requests
+from airbyte_cdk.sources import AbstractSource
+from airbyte_cdk.sources.streams import Stream
+from airbyte_cdk.sources.streams.http import HttpStream
+from airbyte_cdk.sources.streams.http.auth import TokenAuthenticator
+from airbyte_cdk.models import SyncMode
+
+# Basic full refresh stream
+class ZendeskSellStream(HttpStream, ABC):
+    """
+    This class represents a stream output by the connector.
+    This is an abstract base class meant to contain all the common functionality at the API level e.g: the API base URL, pagination strategy,
+    parsing responses etc..
+    """
+
+    url_base = "https://api.getbase.com/v2/"
+    primary_key = None
+
+    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
+        try:
+            regex_page='[=?/]page[_=/-]?(\d{1,3})'
+            meta_links = response.json().get('meta', {}).get('links')
+            if 'next_page' in meta_links.keys():
+                return {'page' : int(re.findall(regex_page, meta_links['next_page'])[0])}
+        except: 
+            return None
```

Comment:
Please add a unit test for this function using the expected payload and one which would fail.

Reply:
To be honest, it would be better to add a try/catch error because you're using a complex function. If it fail you can output what is the input variables so people can understand better whaat cause the failure.